crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -73,6 +73,20 @@ module sui::validator_set {
         tallying_rule_global_score: u64,
     }
 
+    /// Event emitted every time a new validator joins the committee.
+    /// The epoch value corresponds to the first epoch this change takes place.
+    struct ValidatorJoinEvent has copy, drop {
+        epoch: u64,
+        validator_address: address,
+    }
+
+    /// Event emitted every time a validator leaves the committee.
+    /// The epoch value corresponds to the first epoch this change takes place.
+    struct ValidatorLeaveEvent has copy, drop {
+        epoch: u64,
+        validator_address: address,
```

Comment:
Might be useful to include their staking pool ids too, especially for leaving validators since `inactive_validators` are keyed by pool ids.

Reply:
Right, done!